src/include/detail/ivf/index.h

```diff
@@ -71,13 +73,15 @@ int ivf_index(
   auto parts = detail::flat::qv_partition(centroids, db, nthreads);
   debug_matrix(parts, "parts");
   {
-
     scoped_timer _{"shuffling data"};
+    std::unordered_set<ids_type> deleted_ids_set(deleted_ids.begin(), deleted_ids.end());
     std::vector<size_t> degrees(centroids.num_cols());
     std::vector<ids_type> indices(centroids.num_cols() + 1);
     for (size_t i = 0; i < db.num_cols(); ++i) {
-      auto j = parts[i];
-      ++degrees[j];
+      if (auto iter = deleted_ids_set.find(external_ids[i]); iter == deleted_ids_set.end()) {
+        auto j = parts[i];
+        ++degrees[j];
+      }
     }
     indices[0] = 0;
     std::inclusive_scan(begin(degrees), end(degrees), begin(indices) + 1);
```

Comment:
For an updatable index we should probably be using a data structure that isn't contiguous.

Reply:
I am not sure I understand this comment. The degrees data structure is working as expected as we just filter out the deleted vectors.